packages/cli/src/ldap/__tests__/ldap.service.test.ts

```diff
@@ -0,0 +1,1603 @@
+import type { EventService } from '@/events/event.service';
+import {
+	BINARY_AD_ATTRIBUTES,
+	LDAP_LOGIN_ENABLED,
+	LDAP_LOGIN_LABEL,
+	LDAP_FEATURE_NAME,
+} from '@/ldap/constants';
+import { LdapService } from '@/ldap/ldap.service.ee';
+import type { LdapConfig } from '@/ldap/types';
+import { mockInstance, mockLogger } from '@test/mocking';
+
+import {
+	getLdapIds,
+	createFilter,
+	escapeFilter,
+	resolveBinaryAttributes,
+	processUsers,
+	mapLdapUserToDbUser,
+	saveLdapSynchronization,
+	resolveEntryBinaryAttributes,
+} from '../helpers.ee';
+
+// Mock ldapts client
+jest.mock('ldapts', () => {
+	const ClientMock = jest.fn();
+
+	ClientMock.prototype.bind = jest.fn();
+	ClientMock.prototype.unbind = jest.fn();
+	ClientMock.prototype.startTLS = jest.fn();
+	ClientMock.prototype.search = jest.fn();
+
+	return { Client: ClientMock };
+});
+
+jest.mock('../helpers.ee', () => ({
+	...jest.requireActual('../helpers.ee'),
+	getLdapIds: jest.fn(),
+	saveLdapSynchronization: jest.fn(),
+	resolveBinaryAttributes: jest.fn(),
+	processUsers: jest.fn(),
+	resolveEntryBinaryAttributes: jest.fn(),
+}));
+
+jest.mock('n8n-workflow', () => ({
+	...jest.requireActual('n8n-workflow'),
+	randomString: jest.fn(),
+}));
+
+describe('LdapService', () => {
+	const ldapConfig: LdapConfig = {
+		loginEnabled: true,
+		loginLabel: 'fakeLoginLabel',
+		connectionUrl: 'connection.url',
+		allowUnauthorizedCerts: true,
+		connectionSecurity: 'none',
+		connectionPort: 1234,
+		baseDn: 'dc=example,dc=com',
+		bindingAdminDn: 'uid=jdoe,ou=users,dc=example,dc=com',
+		bindingAdminPassword: 'fakePassword',
+		firstNameAttribute: 'givenName',
+		lastNameAttribute: 'sn',
+		emailAttribute: 'mail',
+		loginIdAttribute: 'uid',
+		ldapIdAttribute: 'uid',
+		userFilter: '(uid=jdoe)',
+		synchronizationEnabled: true,
+		synchronizationInterval: 60,
+		searchPageSize: 1,
+		searchTimeout: 6,
+	};
+
+	beforeAll(() => {
+		// Need fake timers to avoid setInterval
+		// problems with the scheduled sync
+		jest.useFakeTimers();
+	});
+
+	beforeEach(() => {
+		jest.restoreAllMocks();
+	});
+
+	describe('init()', () => {
+		it('should load the LDAP configuration', async () => {
+			const settingsRepository = mock<SettingsRepository>({
+				findOneByOrFail: jest.fn().mockResolvedValue({ value: JSON.stringify(ldapConfig) }),
+			});
+
+			const ldapService = new LdapService(mockLogger(), settingsRepository, mock(), mock());
+
+			const loadConfigSpy = jest.spyOn(ldapService, 'loadConfig');
+
+			await ldapService.init();
+
+			expect(loadConfigSpy).toHaveBeenCalledTimes(1);
+		});
+
+		it('should set expected configuration variables from LDAP config if LDAP is enabled', async () => {
+			const settingsRepository = mockInstance(SettingsRepository, {
+				findOneByOrFail: jest.fn().mockResolvedValue({ value: JSON.stringify(ldapConfig) }),
+			}); // set in container so `setCurrentAuthenticationMethod` does not fail - legacy LDAP code not using DI
```

Comment:
you can move `const settingsRepository = mockInstance(SettingsRepository)` out of all these tests, and replace this line with 
```suggestion
      settingsRepository.findOneByOrFail.calledWith({ key: LDAP_FEATURE_NAME }).mockResolvedValueOnce({ value: JSON.stringify(ldapConfig) });
```

Reply:
Thanks, @netroy. It looks like `calledWith` doesn't yet support objects as I couldn't get it to work. Looks like they don't support object comparisons yet (see https://github.com/marchaos/jest-mock-extended/issues/118). I'll refactor without that for now.